uast/nodes/iter.go

```diff
@@ -0,0 +1,213 @@
+package nodes
+
+import "fmt"
+
+type Iterator interface {
+	// Next advances an iterator.
+	Next() bool
+	// Node returns a current node.
+	Node() External
+}
+
+var _ Iterator = Empty{}
+
+type Empty struct{}
+
+func (Empty) Next() bool     { return false }
+func (Empty) Node() External { return nil }
+
+type IterOrder int
+
+const (
+	IterAny = IterOrder(iota)
+	PreOrder
+	PostOrder
+	LevelOrder
```

Comment:
`PositionOrder` missing here.

Reply:
It's intentional - this package cannot define this iterator since all `Position`-related definitions are in the `uast` package. This enum is reflected into `query` package that has all kinds of iterators available.